MediaBrowser.Model/Dlna/StreamBuilder.cs

```diff
@@ -386,32 +387,33 @@ private static TranscodeReason GetTranscodeReasonForFailedCondition(ProfileCondi
         /// <param name="type">The <see cref="DlnaProfileType"/>.</param>
         /// <param name="playProfile">The <see cref="DirectPlayProfile"/> object to get the video stream from.</param>
         /// <returns>The normalized input container.</returns>
-        public static string? NormalizeMediaSourceFormatIntoSingleContainer(string inputContainer, DeviceProfile? profile, DlnaProfileType type, DirectPlayProfile? playProfile = null)
+        public static string NormalizeMediaSourceFormatIntoSingleContainer(string inputContainer, DeviceProfile? profile, DlnaProfileType type, DirectPlayProfile? playProfile = null)
         {
-            if (string.IsNullOrEmpty(inputContainer))
+            if (!inputContainer.Contains(',', StringComparison.OrdinalIgnoreCase) || profile == null)
             {
-                return null;
+                return inputContainer;
             }
 
-            var formats = ContainerProfile.SplitValue(inputContainer);
-
-            if (profile is not null)
+            var formats = inputContainer.SpanSplit(',');
+            var playProfiles = playProfile is null ? profile.DirectPlayProfiles : new[] { playProfile };
```

Comment:
Since you want to get rid of `new[] { ... }`.
```suggestion
            var playProfiles = playProfile is null ? profile.DirectPlayProfiles : [playProfile];
```